src/repr/src/row/encoding2.rs

```diff
@@ -480,7 +508,29 @@ impl DatumColumnEncoder {
             DatumColumnEncoder::I64(mut builder) => Arc::new(builder.finish()),
             DatumColumnEncoder::F32(mut builder) => Arc::new(builder.finish()),
             DatumColumnEncoder::F64(mut builder) => Arc::new(builder.finish()),
-            DatumColumnEncoder::Numeric(mut builder) => Arc::new(builder.finish()),
+            DatumColumnEncoder::Numeric {
+                mut approx_values,
+                mut binary_values,
+            } => {
+                let approx_array = approx_values.finish();
+                let binary_array = binary_values.finish();
+
+                assert_eq!(approx_array.len(), binary_array.len());
+                assert_eq!(approx_array.logical_nulls(), binary_array.logical_nulls());
```

Comment:
This looks like it's `O(n)`... maybe this should be a debug assert?

Reply:
Good call! Updated!